tests/browser/index.test.js

```diff
@@ -310,6 +310,35 @@ suite('<Entity events/>', () => {
     });
   });
 
+  test('can remove event listener when component is detached', function (done) {
+    let bazHandlerCalled = false;
+    function bazHandler () { bazHandlerCalled = true }
+
+    ReactDOM.render(
+      <Scene>
+        <Entity id='foo'>
+          <Entity id='bar' events={{baz: bazHandler}}/>
+        </Entity>
+      </Scene>,
+      div
+    );
+    div.querySelector('a-scene').addEventListener('loaded', () => {
+      ReactDOM.render(
+        <Scene>
+          <Entity id='foo'/>
+        </Scene>,
+        div
+      );
+      div.querySelector('a-entity').emit('baz');
```

Comment:
What I don't understand from the test, is:

> div.querySelector('a-entity').emit('baz');

The query selector will return the `#foo` Entity, and you're emitting `baz` on that. But the `baz` handler was registered on the `#bar` Entity. So emitting `baz` on `#foo` should never do anything. Not sure if this asserts anything useful.

Can you describe the steps of what happens to yield this bug? (e.g., 1. Entity is detached, 2. Event is emitted on...)